File: EcoQuestDesktop/EcoQuestDesktop/Models/Reto.cs
using CommunityToolkit.Mvvm.ComponentModel;
using Newtonsoft.Json;
using System.Runtime.Serialization;

namespace EcoQuestDesktop.Models
{
    [DataContract]
    public partial class Reto : ObservableObject
    {
        [ObservableProperty]
        [JsonProperty("id")]
        [DataMember]
        private int _id;

        [ObservableProperty]
        [JsonProperty("nombre")]
        [DataMember]
        private string _nombre = string.Empty;

        [ObservableProperty]
        [JsonProperty("descripcion")]
        [DataMember]
        private string _descripcion = string.Empty;

        [ObservableProperty]
        [JsonProperty("puntos")]
        [DataMember]
        private int _puntos;

        [ObservableProperty]
        [JsonProperty("tipo")]
        [DataMember]
        private string _tipo = string.Empty;

        [ObservableProperty]
        [JsonProperty("requisitoCantidad")]
        [DataMember]
        private int _requisitoCantidad = 1;

        [ObservableProperty]
        [JsonProperty("frecuencia")]
        [DataMember]
        private string _frecuencia = "ILIMITADA";

        public string FrecuenciaLabel => Frecuencia switch
        {
            "DIARIA" => "Diario",
            "UNICA" => "Una vez",
            "ILIMITADA" => "Ilimitado",
            _ => Frecuencia
        };

        public Reto() { }
    }
}

File: EcoQuestMobile/app/src/main/java/com/ecoquest/app/domain/model/Reto.kt
package com.ecoquest.app.domain.model

data class Reto(
    val id: Long = 0,
    val nombre: String = "",
    val descripcion: String = "",
    val puntos: Int = 0,
    val tipo: String = "",
    val requisitoCantidad: Int = 1,
    val frecuencia: String = "ILIMITADA"
) {
    val frecuenciaLabel: String
        get() = when (frecuencia) {
            "DIARIA" -> "Diario"
            "UNICA" -> "Una vez"
            "ILIMITADA" -> "Ilimitado"
            else -> frecuencia
        }
}
